Enable resize_images_with_crop_or_pad to pad with normal tensorflow.pad calls (mode, name, and constant_values)

brianmanderson opened the pull request:
I was wanting to resize_with_crop_or_pad and pad with values not equal to 0. Now you can pass the same arguments that will feed into pad into resize_with_crop_or_pad

brianmanderson committed:
Update image_ops_impl.py

Passing mode, name, and constant_values from pad_to_bounding_box to pad

brianmanderson committed:
Update image_ops_impl.py

Pass mode, name, constant_values from resize_images_with_crop_or_pad to pad_to_bounding_box which passes to pad

brianmanderson commented:
@googlebot I signed it!

hyeygit reviewed (commented):
It would be nice to have this additional functionality. However, because it changes the API, we need to make sure to update the docstring (and examples in it) to reflect the changes. For example, wherever it says "pads with zero" is now only true with `mode="CONSTANT"` and `constant_values=0`. For both functions `pad_to_bounding_box` and `resize_image_with_crop_or_pad`, could you please update the docstrings and examples (e.g. https://github.com/tensorflow/tensorflow/blob/e0266dbf39deac09315b764524835299b513926c/tensorflow/python/ops/array_ops.py#L3300-L3319) to reflect the changes?

brianmanderson commented:
Sure, I'll start updating it and will @ you when it is finished
Thanks!

brianmanderson committed:
Update image_ops_impl.py

Updated example within the code

brianmanderson commented:
@hyeygit Sorry for the delay, I have updated the example ^^

brianmanderson committed:
Update image_ops_impl.py

corrected spelling and pathing*

brianmanderson committed:
Update image_ops_impl.py

misspelled SYMMETRIC

hyeygit commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -1063,7 +1063,7 @@ def crop_to_bounding_box(image, offset_height, offset_width, target_height,
     v1=['image.resize_with_crop_or_pad', 'image.resize_image_with_crop_or_pad'])
 @dispatch.add_dispatch_support
-def resize_image_with_crop_or_pad(image, target_height, target_width):
+def resize_image_with_crop_or_pad(image, target_height, target_width, mode="CONSTANT", name=None, constant_values=0):
```
There are two places in the docstring that need to be updated. Where it says `"...padding it evenly with zeros"` and `"...this op centrally pads with 0 along..."`. Could you update the description please?

brianmanderson replied:
Sorry, this has been resolved to properly reference 'constant_values' instead of 0

hyeygit commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -1080,6 +1080,10 @@ def resize_image_with_crop_or_pad(image, target_height, target_width):
       of shape `[height, width, channels]`.
     target_height: Target height.
     target_width: Target width.
+    mode: One of "CONSTANT", "REFLECT", or "SYMMETRIC" (case-insensitive)
```
Could we please append a description to `mode` pointing users to the description / example in `pad_to_bounding_box`? e.g. "See examples in `pad_to_bounding_box`."

brianmanderson replied:
I have included a line pointing users to find description and examples in 'pad_to_bounding_box'

hyeygit reviewed (commented):
Thank you so much for addressing the previous comments! I reviewed the new changes and left a few comments.

gbaned commented:
@brianmanderson Can you please check @hyeygit's comments and keep us posted. Thanks!

brianmanderson committed:
Update image_ops_impl.py

Updated pad_to_bounding_box to properly reference 'constant_values' and not just 0

brianmanderson committed:
Update image_ops_impl.py

updated example to more easily interpreted one provided by hyeygit

brianmanderson committed:
Update image_ops_impl.py

corrected doc_string to reference 'constant_values', not 0

brianmanderson committed:
Update image_ops_impl.py

updated to point to examples in 'pad_to_bounding_box' in 'resize_image_with_crop_or_pad'

brianmanderson commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -878,38 +878,38 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
-  [ 0.,  0.,  0.],
-  [ 0.,  0.,  0.]]], dtype=float32)>
-
+  t = tf.constant([[1, 2, 3], [4, 5, 6]])
```
thank you for the example, it is not included ^^

brianmanderson commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -867,7 +867,7 @@ def _get_dim(tensor, idx):
 def pad_to_bounding_box(image, offset_height, offset_width, target_height,
-                        target_width):
+                        target_width, mode="CONSTANT", name=None, constant_values=0):
   """Pad `image` with zeros to the specified `height` and `width`.
```
this has been fixed to refer to 'constant_values', not 0, sorry for not updating it on the first iteration

brianmanderson commented:
@gbaned Absolutely ^^, I've updated the fork to reflect @hyeygit 's changes and recommendations
Thank you!

hyeygit reviewed (dismissed):
Thank you for making the changes!

gbaned commented:
@brianmanderson Can you please address Ubuntu Sanity errors? Thanks!

gbaned commented:
@brianmanderson Any update on this PR? Please. Thanks!

brianmanderson commented:
Hello, sorry for the delay, I will address this today ^^

brianmanderson committed:
Update image_ops_impl.py

Update sentence length to fix non-allowed list pylint errors

brianmanderson commented:
@gbaned updated, thank you ^^

brianmanderson committed:
Update image_ops_impl.py

trying to pass the pylint sanity error that I have lines too long

brianmanderson commented:
@gbaned @hyeygit Hello, I reviewed the sanity errors again today, looks like I keep failing to fit everything within the allotted amount of line space.. I think that everything should pass the pylinting now
Thank you for your patience
Brian

gbaned commented:
@brianmanderson Still, build failures are appearing, Can you fix those?. Thank you!

gbaned commented:
@brianmanderson Any update on this PR? Please. Thanks!

gbaned commented:
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!
